examples/david_map_reduce_weighted.py

```diff
@@ -158,52 +155,42 @@ def main(
     # Compute a coreset using kernel herding with a Stein kernel. To reduce compute
     # time, we apply MapReduce, which partitions the input into blocks for independent
     # coreset solving. We also reduce memory requirements by specifying block size
-    herding_key, sample_key = random.split(random.key(random_seed))
-    herding_object = KernelHerding(
-        herding_key,
+    _, sample_key = random.split(random.key(random_seed))
+    herding_solver = KernelHerding(
+        coreset_size,
         kernel=herding_kernel,
-        weights_optimiser=weights_optimiser,
         block_size=1_000 // downsampling_factor,
     )
-    herding_object.fit(
-        original_data=data,
-        strategy=MapReduce(
-            coreset_size=coreset_size, leaf_size=10_000 // downsampling_factor
-        ),
+    mapped_herding_solver = MapReduce(
+        herding_solver, leaf_size=10_000 // downsampling_factor
     )
-    herding_weights = herding_object.solve_weights()
+    herding_coreset, _ = eqx.filter_jit(mapped_herding_solver.reduce)(data)
+    herding_weights = weights_optimiser.solve(data, herding_coreset.coreset)
 
     print("Choosing random subset...")
     # Generate a coreset via uniform random sampling for comparison
-    random_sample_object = RandomSample(sample_key, unique=True)
-    random_sample_object.fit(
-        original_data=data,
-        strategy=SizeReduce(coreset_size=coreset_size),
-    )
+    random_solver = RandomSample(coreset_size, sample_key, unique=True)
+    random_coreset, _ = eqx.filter_jit(random_solver.reduce)(data)
```

Comment:
As above comment; see here https://github.com/gchq/coreax/pull/632#discussion_r1620424830

Reply:
It might be worth documenting this somewhere, although the examples could be documentation enough themselves. What do you think for a user trying to use the library for the first time?

Reply:
I'm happy to explain why we are doing this in the `quickstart.rst` part of the docs

Reply:
@tc85324 I'd be happy with that, thanks